pkg/util/util.go

```diff
@@ -22,3 +29,31 @@ func StringMatchRegexSlice(s string, re []string) bool {
 	}
 	return false
 }
+
+// ReadPassword reads a password from the terminal and restores the terminal state on interruption
+func ReadPassword() (string, error) {
+	var err error
+	var password []byte
+	// Store and restore the terminal status on interruptions to
+	// avoid that the terminal remains in the password state
+	// This is necessary as a workaround for https://github.com/golang/go/issues/31180
+	stdin := int(os.Stdin.Fd())
+	oldState, err := terminal.GetState(stdin)
+	if err != nil {
+		oldState = nil
+	}
+	sigch := make(chan os.Signal, 1)
+	defer close(sigch)
+	signal.Notify(sigch, os.Interrupt)
+	go func() {
+		for range sigch {
+			if oldState != nil {
+				terminal.Restore(stdin, oldState)
+			}
+			err = fmt.Errorf("interrupted")
```

Comment:
AFAICS this doesn’t work at all: The original tiny example has an `os.Exit`, which we don’t want.

But doing _nothing_ here would not interrupt `terminate.ReadPassword` at all; in my experiments the program just starts echoing Ctrl-C (because of the `.Restore`) but `ReadPassword` never exits. And I can’t see any plausible way to terminate `ReadPassword`, except maybe for closing the `stdin` FD, which would just result in mayhem (because the FD would be reassigned by any future file open)

Reply:
This is very true. Please see my comment below for advice on how to proceed.